### DISCUSSION TITLE
Microsoft Auth Provider - User Name and email are lacking in pocketbase?

### DISCUSSION THREAD

**Discussion by jpetrov** [pocketbase/pocketbase#4764] [Q&A] [answered]
Sorry to address this, it might be a noob question, but unlike Google and Facebook integrations where the user is referenced to its email address, the Microsoft sign-on works ok. Still, it creates new users who can re-authenticate to the system, but the email and the user's full name are not filled in. If I use Facebook's sign-on with the same email as Google's, the login goes to the same profile as it should, but Microsoft sign-on creates a new user without key data. If I set the Always require email, the Microsoft logon fails, and it does not proceed to the active session.

<img width="842" alt="Screenshot 2024-04-16 at 12 48 29" src="https://github.com/pocketbase/pocketbase/assets/1609500/e9760071-a8d2-41bb-8655-54c907e78960">

**Comment by ganigeorgiev** [comment, marked as answer]
The Microsoft OAuth2 provider doesn't return a username field so we set an auto generated one.

For the email - make sure that your Microsft OAuth2 app is properly configured. You can check the notes provided by other users in:

- https://github.com/pocketbase/js-sdk/issues/176#issuecomment-1518939458
- https://github.com/pocketbase/pocketbase/discussions/3371#discussioncomment-7116631
- https://github.com/pocketbase/pocketbase/discussions/3612#discussioncomment-7402195

**Reply by ganigeorgiev** [reply]
On second read, you are not asking for the `username` field but for the `name` one, right?

If that's the case then you need to send an additional update request with the response field value from the OAuth2 (the OAuth2 user data are returned under the `meta` response object).

This secondary update request for the `name` field is required because we currently only map automatically the `email` (if it is available) and the `username` (if it is available and matches the PocketBase `username` field requirements).

_With the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) there will be an optional collection configuration that will allow explicitly specifying the mapped fields from the Admin UI (including automatically uploading the `avatarUrl` file to a `file` field)._

**Reply by jpetrov** [reply]
thanks, it all makes sense now, i have more homework with MPN as I didn't know that...
